src/main/scala/io/pravega/connectors/spark/PravegaMicroBatchStream.scala

```diff
@@ -70,13 +80,47 @@ class PravegaMicroBatchStream(
   }
   log.info(s"resolvedStartStreamCut=${resolvedStartStreamCut}, resolvedEndStreamCut=${resolvedEndStreamCut}")
 
+  if (maxEventPerTrigger.isDefined) {
+      readerGroupManager = ReaderGroupManager.withScope(scopeName, clientConfig.getControllerURI())
+      readerGroupManager.createReaderGroup(readerGroupName, ReaderGroupConfig.builder().stream(Stream.of(scopeName, streamName), resolvedStartStreamCut).build())
+      readerGroup = readerGroupManager.getReaderGroup(readerGroupName)
+      clientFactory = EventStreamClientFactory.withScope(scopeName, clientConfig)
+      eventReader = clientFactory.createReader("reader", readerGroupName, new ByteBufferSerializer, ReaderConfig.builder().build())
+  }
+
+  override def getDefaultReadLimit: ReadLimit = {
+    if (maxEventPerTrigger.isDefined) {
+      ReadLimit.maxRows(maxEventPerTrigger.get)
+    } else {
+      super.getDefaultReadLimit
+    }
+  }
 
   override def initialOffset(): Offset = {
     PravegaSourceOffset(resolvedStartStreamCut)
   }
 
   override def latestOffset(): Offset = {
-    PravegaSourceOffset(streamManager.getStreamInfo(scopeName, streamName).getTailStreamCut)
+    throw new UnsupportedOperationException(
+      "latestOffset(Offset, ReadLimit) should be called instead of this method")
+  }
+
+  override def latestOffset(start: Offset, readLimit: ReadLimit): Offset = {
+    if(readLimit.isInstanceOf[ReadAllAvailable]) {
+      PravegaSourceOffset(streamManager.getStreamInfo(scopeName, streamName).getTailStreamCut)
+    } else {
+      val maxRows = readLimit.asInstanceOf[ReadMaxRows].maxRows()
+      var event: EventRead[ByteBuffer] = null
+      var count: Long = 0
+      do {
+        event = eventReader.readNextEvent(100)
```

Comment:
This will cause the driver to read all events from Pravega. We need to avoid that.